lib/blob.js

```diff
@@ -359,19 +604,496 @@ Blob.prototype.getContainerMetadata = function getContainerMetadata(name) {
+      }
+      if (options.include.metadata) {
+        includeValue += 'metadata%82';
+      }
+      if (options.include.uncommittedBlobs) {
+        includeValue += 'uncommittedBlobs%82';
+      }
+      if (options.include.copy) {
+        includeValue += 'copy%82';
+      }
+      query.include = includeValue;
+    }
+    if (options.delimiter)  query.delimiter = options.delimiter;
+  }
+
+  var path = '/' + container;
+  var headers = {};
+
+  return this.request('GET', path, query, headers).then(function(response){
+    if(response.statusCode !== 200){
+      throw new Error("setContainerACL: Unexpected statusCode: " + response.statusCode);
+    }
+    return xml.blobParseListBlobs(response);
+  });
+};
+
+/**
+ * Establishes and manages a lock on a container for delete operations. The lock duration can be 15 to 60 seconds, or can be infinite.
+ *
+ * @method leaseContainer
+ * @param name - Name of the container
+ * @param {object} options - Options on the following form
+ * ```js
+ * {
+ *    leaseId: '...',           // GUID string; it is required in case of renew, change, or release of the lease.
+ *    leaseAction: '...',       // Lease container operation. The possible values are: acquire, renew, change, release, break (required)
+ *    leaseBreakPeriod: '...',  // For a break operation, proposed duration the lease should continue before it is broken, in seconds, between 0 and 60.
+ *    leaseDuration: '...',     // Specifies the duration of the lease, in seconds, or negative one (-1) for a lease that never expires.
+ *                              // Required for acquire.
+ *    proposedLeaseId: '...'    // GUID string; Optional for acquire, required for change.
+ * }
+ * ```
+ * @returns {Promise} A promise for an object on the form:
+ * ```js
+ * {
+ *    leaseId: '...',   // The unique lease id.
+ *    leaseTime: '...'  // Approximate time remaining in the lease period, in seconds.
+ * }
+ * ```
+ */
+Blob.prototype.leaseContainer = function leaseContainer(name, options) {
+  assert(name, 'The name of the container must be specified');
+  var query = {
+    restype: 'container',
+    comp: 'lease'
+  };
+
+  var path = '/' + name;
+  var headers = {};
+
+  assert(options.leaseAction, "The `options.leaseAction` must be given");
+
+  if (options.leaseId) {
+    assert(utils.isValidGUID(options.leaseId), 'The supplied `leaseId` is not a valid GUID');
+    headers['x-ms-lease-id'] = options.leaseId;
+  }
+
+  assert(options.leaseAction === 'acquire' || options.leaseAction === 'renew' || options.leaseAction === 'change' || options.leaseAction === 'release' || options.leaseAction === 'break',
+    'The supplied `options.leaseAction` is not valid. The possible values are: acquire, renew, change, release, break');
+  headers['x-ms-lease-action'] = options.leaseAction;
+
+  if((options.leaseAction === 'renew' || options.leaseAction === 'change' || options.leaseAction === 'release') && !options.leaseId) {
```

Comment:
these are super-cheap operations in JS, especially when compared to the overhead of making an HTTP request.  I think the readability is more important than a few ms.  But line breaks would be an improvement.

Either line breaks or `.indexOf()` is fine with me.